### DISCUSSION TITLE
assert introspection despite of custom message

### DISCUSSION THREAD

**Discussion by pmeier** [pytest-dev/pytest#8426] [Q&A] [answered]
The documentation [states](https://docs.pytest.org/en/stable/assert.html#asserting-with-the-assert-statement)

> However, if you specify a message with the assertion like this:
> 
> `assert a % 2 == 0, "value was odd, should be even"`
>
> then no assertion introspection takes places at all and the message will be simply shown in the traceback.

Running

```python
def foo():
    return 1


def test_foo():
     assert foo() % 2 == 0, "value was odd, should be even"
```

gives the following output

```
________________________________________________ test_foo _________________________________________________

    def test_foo():
>        assert foo() % 2 == 0, "value was odd, should be even"
E        AssertionError: value was odd, should be even
E        assert (1 % 2) == 0
E         +  where 1 = foo()
```

Given that there is introspection, I'm either misinterpreting the documentation or it is wrong / outdated. Could someone tell me which one is the case?

Bonus: Can someone tell me how to disable this introspection locally instead going for the [whole module or test suite](https://docs.pytest.org/en/stable/assert.html#disabling-assert-rewriting)?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
i believe `--assert=plain` should do the trick of disable

wrt the docs, at first glance i believe its misleading there,
my understanding is that assertrepr_compare is disabled but not all inspection

i'll have to crosscheck later when i find some more time, for clarity we should include a example output that makes sense in any case

**Reply by RonnyPfannschmidt** [reply]
whops, i did misread a bit, currently you have to go for the whole testsuite

**Reply by pmeier** [reply]
While it is not helpful in my case, you can [disable introspection on a module level](https://docs.pytest.org/en/stable/assert.html#disabling-assert-rewriting):

> Disable rewriting for a specific module by adding the string `PYTEST_DONT_REWRITE` to its docstring.
